### DISCUSSION TITLE
[ZK Stack] - can't get pass ```zkstack ecosystem init --dev``` step

### DISCUSSION THREAD

**Discussion by rithythul** [zkSync-Community-Hub/zksync-developers#793] [ZK Stack] [answered]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [ ] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.82.0

### Operating System

- [ ] Ubuntu
- [ ] MacOS
- [ ] Windows
- [X] Other (Please specify below)

### Issue Description

Error when running ```zkstack ecosystem init --dev --verbose ```

### Expected Behavior

This is my commands history 
```
curl -L https://raw.githubusercontent.com/matter-labs/foundry-zksync/main/install-foundry-zksync | bash
zkstackup 
zkstack ecosystem create
cd my_elastic_chain/
zkstack ecosystem init --dev --verbose 

```

I have gone through this many times with a few iteration. 

I also install libclang ```sudo pacman -S llvm`` 

### Current Behavior

   Compiling itertools v0.11.0
error: failed to run custom build command for `librocksdb-sys v0.11.0+8.1.1`

Caused by:
  process didn't exit successfully: `/home/pi/my_elastic_chain/zksync-era/target/release/build/librocksdb-sys-f7fa57978d111495/build-script-build` (exit status: 101)
  --- stderr
  thread 'main' panicked at /home/pi/.cargo/registry/src/index.crates.io-6f17d22bba15001f/bindgen-0.65.1/lib.rs:603:31:
  Unable to find libclang: "couldn't find any valid shared libraries matching: ['libclang.so', 'libclang-*.so', 'libclang.so.*', 'libclang-*.so.*'], set the `LIBCLANG_PATH` environment variable to a path where one of these files can be found (invalid: [])"
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
│
■  Command failed to run
│  
■    Status:
│      exit status: 101
│    Stdout:
│      
│  
│    Stderr:
│      
│  
■  Unable to perform genesis on the database
│  
▲    0: Failed to start server
│    1: Command failed to run: cargo run --release --bin zksync_server -- --genesis-path /home/pi/my_elastic_chain/chains/zk_chain_1/configs/genesis.yaml --wallets-path /home/pi/my_elastic_chain/chains/zk_chain_1/configs/wallets.yaml --config-path /home/pi/my_elastic_chain/chains/zk_chain_1/configs/general.yaml --secrets-path /home/pi/my_elastic_chain/chains/zk_chain_1/configs/secrets.yaml --contracts-config-path /home/pi/my_elastic_chain/chains/zk_chain_1/configs/contracts.yaml --genesis
│  
│    2: Command failed to run: cargo run --release --bin zksync_server -- --genesis-path /home/pi/my_elastic_chain/chains/zk_chain_1/configs/genesis.yaml --wallets-path /home/pi/my_elastic_chain/chains/zk_chain_1/configs/wallets.yaml --config-path /home/pi/my_elastic_chain/chains/zk_chain_1/configs/general.yaml --secrets-path /home/pi/my_elastic_chain/chains/zk_chain_1/configs/secrets.yaml --contracts-config-path /home/pi/my_elastic_chain/chains/zk_chain_1/configs/contracts.yaml --genesis
│  
└  Failed to run command

### Additional Details

The chains seem to run in the docker 
![seastar](https://github.com/user-attachments/assets/2da1e859-8f0f-4de1-abb7-8a324188d6fb)

### Prior Research

- [x] Yes, I searched for similar issues and did not find any matching ones.

**Comment by alhenaai** [comment]
It appears you are encountering a build error related to `librocksdb-sys` when trying to initialize the ZKsync ecosystem on your system. The error message indicates that `libclang` is not found, which is necessary for the `bindgen` tool used in the build process.

Here are a few steps to resolve this issue:

1. **Set the `LIBCLANG_PATH` Environment Variable:**
   Ensure that `libclang` is installed on your system and set the `LIBCLANG_PATH` environment variable to the directory containing `libclang.so`. Since you are using Arch Linux (as indicated by the use of `pacman`), you might need to locate the `libclang.so` file (usually found in `/usr/lib/`) and set the environment variable accordingly:
   ```bash
   export LIBCLANG_PATH=/usr/lib/
   ```

2. **Verify `libclang` Installation:**
   Double-check that `libclang` is properly installed. You can reinstall it to make sure:
   ```bash
   sudo pacman -S clang
   ```

3. **Re-run the Initialization:**
   After setting up `LIBCLANG_PATH`, try running the `zkstack ecosystem init --dev --verbose` command again.

4. **Check Dependencies:**
   Ensure all other dependencies are correctly installed and compatible with your setup, especially since you are running this on a Raspberry Pi, which might have different compatibility considerations compared to typical x86_64 architectures.

If the problem persists after these steps, consider checking the compatibility of the libraries and the build tools with the ARM architecture of the Raspberry Pi, as this can sometimes cause issues not seen on other architectures.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Reply by rithythul** [reply]
@dimazhornyk here is my error repoort

**Reply by alhenaai** [reply]
Thank you for your question! I’m unable to provide an answer at this moment, but I have forwarded your inquiry to our team. Someone will be with you shortly to assist further. Your patience is appreciated!

**Reply by dimazhornyk** [reply, marked as answer]
Did you try `pacman -S clang`?

**Reply by rithythul** [reply]
I have not tried. Will do in an hour.

**Reply by rithythul** [reply]
Hi @sachanayush47 @dimazhornyk, thanks so much. it works now. So arch Linux need ```clang``` to work. 

Also, i have a small PR about zkstackup which after successfully installed it added ```//``` in of $HOME/.local//bin, which should be $HOME/.local/bin. Not sure if it is because of my pc but I changed the bash script and made the PR.

**Comment by sachanayush47** [comment]
What OS are you using? It looks like your environment is not properly setup.

**Reply by rithythul** [reply]
I am using Arch Linux. 

Can you guide how to properly set up the environment, please?
